src/Models/ShallowWaterModels/update_state.jl

```diff
@@ -0,0 +1,19 @@
+using Oceananigans.Architectures
+using Oceananigans.BoundaryConditions
+
+import Oceananigans.TimeSteppers: update_state!
+
+"""
+    update_state!(model)
+
+Update peripheral aspects of the model (halo regions, diffusivities, hydrostatic pressure) to the current model state.
+"""
+function update_state!(model::ShallowWaterModel)
+
+    # Fill halos for velocities and tracers
+    fill_halo_regions!(merge(model.velocities, model.tracers), model.architecture, 
```

Comment:
🚀

Reply:
I changed velocities in line 14 to solution, since that's the structure.